src/sidebar.js

```diff
@@ -1,386 +1,362 @@
+    WebkitTransition: '-webkit-transform .3s ease-out',
+    willChange: 'transform',
+    overflowY: 'auto'
   },
   content: {
-    position: "absolute",
+    position: 'absolute',
     top: 0,
     left: 0,
     right: 0,
     bottom: 0,
-    overflowY: "auto",
-    WebkitOverflowScrolling: "touch",
-    transition: "left .3s ease-out, right .3s ease-out"
+    overflowY: 'auto',
+    WebkitOverflowScrolling: 'touch',
+    transition: 'left .3s ease-out, right .3s ease-out'
   },
   overlay: {
     zIndex: 1,
-    position: "fixed",
+    position: 'fixed',
     top: 0,
     left: 0,
     right: 0,
     bottom: 0,
     opacity: 0,
-    visibility: "hidden",
-    transition: "opacity .3s ease-out, visibility .3s ease-out",
-    backgroundColor: "rgba(0,0,0,.3)"
+    visibility: 'hidden',
+    transition: 'opacity .3s ease-out, visibility .3s ease-out',
+    backgroundColor: 'rgba(0,0,0,.3)'
   },
   dragHandle: {
     zIndex: 1,
-    position: "fixed",
+    position: 'fixed',
     top: 0,
     bottom: 0
   }
 };
 
-class Sidebar extends Component {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      // the detected width of the sidebar in pixels
-      sidebarWidth: props.defaultSidebarWidth,
-
-      // keep track of touching params
-      touchIdentifier: null,
-      touchStartX: null,
-      touchCurrentX: null,
-
-      // if touch is supported by the browser
-      dragSupported: false
-    };
-
-    this.overlayClicked = this.overlayClicked.bind(this);
-    this.onTouchStart = this.onTouchStart.bind(this);
-    this.onTouchMove = this.onTouchMove.bind(this);
-    this.onTouchEnd = this.onTouchEnd.bind(this);
-    this.onScroll = this.onScroll.bind(this);
-    this.saveSidebarRef = this.saveSidebarRef.bind(this);
-  }
+const Sidebar = props => {
+  // the detected width of the sidebar in pixels
+  const [sidebarWidth, setSidebarWidth] = useState(props.defaultSidebarWidth);
 
-  componentDidMount() {
-    const isIos = /iPad|iPhone|iPod/.test(navigator ? navigator.userAgent : "");
-    this.setState({
-      dragSupported:
-        typeof window === "object" && "ontouchstart" in window && !isIos
-    });
-    this.saveSidebarWidth();
-  }
+  // keep track of touching params
+  const [touchParams, setTouchParams] = useState({
+    identifier: null,
+    startX: null,
+    currentX: null
+  });
+
+  // if touch is supported by the browser
+  const [dragSupported, setDragSupported] = useState(false);
+
+  const sidebarRef = useRef();
+
+  useEffect(() => {
+    const isIos = /iPad|iPhone|iPod/.test(navigator ? navigator.userAgent : '');
+    setDragSupported(
+      typeof window === 'object' && 'ontouchstart' in window && !isIos
+    );
+    saveSidebarWidth();
+  }, []);
 
-  componentDidUpdate() {
+  useEffect(() => {
     // filter out the updates when we're touching
-    if (!this.isTouching()) {
-      this.saveSidebarWidth();
+    if (!isTouching()) {
+      saveSidebarWidth();
     }
-  }
+  });
```

Comment:
Should we try to call this hook only when `touchParams.identifier` changes, could be something like this:

```js
    useEffect(() => {
       saveSidebarWidth();
    }, [touchParams.identifier]);
```